### PR TITLE
Add React Error boundary#13576

### PR DESCRIPTION
**Pull request by Clark-W-Griswold**
## Description
Adding reusable ErrorBoundary component. Adding initial implementation to current RoutentSavableReviewPage. Adding Sentry's react error library. 

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/utilities/error-boundary/index.jsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  const { children, fallback, showDialog = false } = props;
+
+  return (
+    <Sentry.ErrorBoundary
+      fallback={fallback || FallbackComponent}
+      showDialog={showDialog}
+    >
+      {children}
+    </Sentry.ErrorBoundary>
+  );
```
Comment (bkjohnson) [inline src/platform/utilities/error-boundary/index.jsx:20]: So what exactly is this doing for us?

Comment (Clark-W-Griswold) [reply src/platform/utilities/error-boundary/index.jsx:20]: Starting to leverage error boundaries in react. This is Sentry's wrapper around it. 
https://reactjs.org/docs/error-boundaries.html

File: src/platform/utilities/error-boundary/index.jsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  return <div />;
+}
+
+function ErrorBoundary(props) {
+  const { children, fallback, showDialog = false } = props;
+
+  return (
+    <Sentry.ErrorBoundary
+      fallback={fallback || FallbackComponent}
+      showDialog={showDialog}
```
Comment (rianfowler) [inline src/platform/utilities/error-boundary/index.jsx:16]: what does this look like?

**Review by rianfowler — COMMENTED** [review]
Do you anticipate the forms library having multiple error boundaries or moving this out further out as we validate this approach?

**Pull request closed without merging**